airflow/models/taskinstance.py

```diff
@@ -1140,7 +1140,11 @@ def next_retry_datetime(self):
             # of tries (around 50 if the initial delay is 1s, even fewer if
             # the delay is larger). Cap the value here before creating a
             # timedelta object so the operation doesn't fail.
-            delay_backoff_in_seconds = min(modded_hash, timedelta.max.total_seconds() - 1)
+            # the timedelta is limited to core.max_retry_delay_in_seconds config value
+            # or 24 Hrs if not specified
+            delay_backoff_in_seconds = min(
+                modded_hash, conf.getint("core", "default_max_retry_delay", fallback=24 * 60 * 60) - 1
```

Comment:
Looking at this again, we might be better off doing this like default_retry_delay is done in BaseOperator? Certainly more consistent.